src/lsqecc/logical_lattice_ops/logical_lattice_ops.py

```diff
@@ -118,8 +118,10 @@ def to_lattice_operation(
                 self.ops.append(current_op)
 
     def circuit_to_patch_measurement(
-        self, m: Measurement
+        self, m: PauliProductOperation
     ) -> Union[SinglePatchMeasurement, MultiBodyMeasurement]:
+        if not isinstance(m, Measurement):
+            raise TypeError("Make sure the passed argument is of type Measurement")
```

Comment:
was `mypy` failing to enforce this?

Reply:
tests/logical_lattice_ops/logical_lattice_ops_test.py:143: error: Argument 1 to "circuit_to_patch_measurement" of "LogicalLatticeComputation" has incompatible type "PauliProductOperation"; expected "Measurement".

mypy was raising this exception. The argument was type-specified to Measurement. However, the ops in a PauliOpCircuit are of the the supertype PauliProductOperation. So I changed the type specification to PauliProductoperation and internally added a type check.

Reply:
you should be able to use `cast` to work around it

Reply:
Will keep in mind next time. Should I rewrite this bit now ?